### PR TITLE
feat: Telegram group management

### LINKED ISSUE #65: Telegram Group Management ($1,800)
## Overview
Implement capabilities for managing Telegram groups and channels, providing advanced moderation and administration capabilities.

## Core Features
- Member management (add, remove, restrict)
- Permission management
- Content moderation
- Spam protection
- Group settings management
- Channel post management
- Admin action logging

## Acceptance Criteria
- [ ] Complete group management prisms
- [ ] Permission system implementation
- [ ] Moderation action handlers
- [ ] Admin logging system
- [ ] Documentation and examples
- [ ] Test coverage > 90%
- [ ] Performance optimization

## Budget: $1,800

### PR DESCRIPTION
**Pull request by DicksonWu654**
# Telegram Group Management Features Implementation

This PR implements advanced group and channel management capabilities for the Telegram Bot API integration in Lux, providing a complete suite of moderation and administration tools through the lens system.

### Member Management
- **User Control**: Ban, unban, restrict, and promote chat members
- **Permission Granularity**: Fine-grained control over user permissions
- **Admin Management**: Custom admin titles and role assignment
- **Sender Chat Control**: Ban and unban channel chats in groups

### Permission Management
- **Default Permissions**: Set baseline permissions for all members
- **Invite Link System**: Create, edit, and revoke invite links with customization
- **Join Request Handling**: Approve or decline user join requests
- **Access Control**: Manage who can perform specific actions

### Content Moderation
- **Message Management**: Delete individual or multiple messages
- **Media Control**: Delete chat photos and manage media content
- **History Management**: Clean chat history for all users
- **Pinned Content**: Pin, unpin, and manage pinned messages

### Spam Protection
- **Slow Mode**: Configure message rate limiting
- **New Member Restrictions**: Temporarily restrict new members
- **Automated Protection**: Scheduled permission restoration

### Group Settings Management
- **Profile Settings**: Update chat title, description, and photo
- **Visual Identity**: Manage group appearance and information
- **Message Pinning**: Control important announcements

### Channel Post Management
- **Content Editing**: Edit channel post text and captions
- **Post Removal**: Delete channel posts
- **Post Visibility**: Control post distribution

### Admin Action Logging
- **Action Tracking**: Log all administrative actions
- **Dedicated Channels**: Create and configure admin log channels
- **Automated Logging**: Set up automatic logging of all admin actions

### Core Components

- **TelegramGroupManagementFeatures**: Main module providing all group management functionality
  - Complete API method implementations
  - Comprehensive documentation
  - Structured by feature category

- **Error Handling Integration**:
  - Leverages existing error handling system
  - Appropriate error categorization
  - Consistent response formatting

### Example Implementation

- **Complete Group Management Example**:
  - Demonstrates all management features
  - Real-world usage scenarios
  - Comprehensive documentation

- **Feature-Specific Examples**:
  - Member management demonstration
  - Permission system showcase
  - Content moderation examples
  - Spam protection implementation
  - Group settings management
  - Channel post management
  - Admin logging system

## Documentation

- Comprehensive module documentation
- Detailed function documentation with parameters and return values
- Example code snippets for each feature category
- Complete example implementation

## Testing

- **Coverage**: Comprehensive unit tests for all components
- **Verification**: Tests for all API endpoints and parameters
- **Error Handling**: Tests for error conditions and edge cases
- **Integration**: Tests for interaction with other Telegram features

## Performance

- Efficient request handling
- Optimized parameter processing
- Consistent error handling
- Proper resource management

## Related Issues
Closes #65 

This PR builds upon the foundation established in PR #134, extending the Telegram Bot API integration with comprehensive group and channel management capabilities. Thus should be merged after 134. It provides a complete solution for building sophisticated moderation and administration systems in Telegram bots developed with Lux.

### CODE REVIEW

File: lux/lib/lux/lenses/telegram/client.ex [hunk 1]
```diff
@@ -0,0 +1,224 @@
+defmodule Lux.Lenses.Telegram.Client do
```
Comment (rparcus) [inline lux/lib/lux/lenses/telegram/client.ex:1]: Having a client is probably a good idea, but I'd implement it without rate limiting and retry by default. Mention something about this on the other PR as well.
